### PR TITLE
QF-625 Upgrade to .NET 6 and upgrade package versions

### PR DESCRIPTION
**Pull request by SawNawDfE**
Please **_do not merge_** until this ticket is completed: https://skillsfundingagency.atlassian.net/browse/DASD-9851

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Application.Api/SFA.DAS.AssessorService.Application.Api.csproj [hunk 1]
```diff
@@ -1,50 +1,39 @@
...
+    <PackageReference Include="System.ServiceModel.Duplex" Version="4.10.0" />
+    <PackageReference Include="System.ServiceModel.Http" Version="4.10.0" />
+    <PackageReference Include="System.ServiceModel.NetTcp" Version="4.10.0" />
+    <PackageReference Include="System.ServiceModel.Security" Version="4.10.0" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\SFA.DAS.AssessorService.Application\SFA.DAS.AssessorService.Application.csproj" />
     <ProjectReference Include="..\SFA.DAS.AssessorService.ApplyTypes\SFA.DAS.AssessorService.ApplyTypes.csproj" />
     <ProjectReference Include="..\SFA.DAS.AssessorService.Data\SFA.DAS.AssessorService.Data.csproj" />
-    <!--<ProjectReference Include="..\SFA.DAS.AssessorService.Api.Types\SFA.DAS.AssessorService.Api.Types.csproj" />
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.AssessorService.Application.Api/SFA.DAS.AssessorService.Application.Api.csproj:45]: This looks odd The Api.Types are supposed to be all the types which are exposed by the Api, there not due to some bad coding practices but why is the library reference being deleted?

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

**Comment by SawNawDfE** [comment]
Test - ignore

File: src/SFA.DAS.AssessorService.Application.Api/Connected Services/CharityCommissionService/ConnectedService.json [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
+    ],
+    "collectionTypes": [
+      "System.Array",
+      "System.Collections.Generic.Dictionary`2"
+    ],
+    "messageContract": true,
+    "namespaceMappings": [
+      "*, CharityCommissionService"
+    ],
+    "targetFramework": "netcoreapp3.1",
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AssessorService.Application.Api/Connected Services/CharityCommissionService/ConnectedService.json]: The changes look ok to me

File: src/SFA.DAS.AssessorService.Session/SFA.DAS.AssessorService.Session.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>netstandard2.0</TargetFramework>
+        <TargetFramework>netcoreapp3.1</TargetFramework>
```
[the remark this answers is not in the document]
Comment (SawNawDfE) [reply src/SFA.DAS.AssessorService.Session/SFA.DAS.AssessorService.Session.csproj]: This project should no longer exist in the last few commits

File: src/SFA.DAS.AssessorService.Application.Api/SFA.DAS.AssessorService.Application.Api.csproj (hunk 1 above)
Comment (SawNawDfE) [reply src/SFA.DAS.AssessorService.Application.Api/SFA.DAS.AssessorService.Application.Api.csproj:45]: I've added it back in

File: src/SFA.DAS.AssessorService.Application.Api/SFA.DAS.AssessorService.Application.Api.csproj [hunk 4]
```diff
@@ -1,50 +1,39 @@
...
-    <TargetFramework>netcoreapp2.2</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
     <DockerComposeProjectPath>..\docker-compose.dcproj</DockerComposeProjectPath>
     <UserSecretsId>81c72058-1312-4504-b772-231de821dbb7</UserSecretsId>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DocumentationFile>bin\Debug\netcoreapp2.0\SFA.DAS.AssessorService.Application.Api.xml</DocumentationFile>
-    <NoWarn>1591</NoWarn>
+    <DocumentationFile>C:\Source\Repos\SFA\das-assessor-service\src\SFA.DAS.AssessorService.Application.Api\SFA.DAS.AssessorService.Application.Api.xml</DocumentationFile>
+    <NoWarn>1701;1702;1591</NoWarn>
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AssessorService.Application.Api/SFA.DAS.AssessorService.Application.Api.csproj]: I've fixed this by using the default file path - which generated into the net6.0 debug or release directories

**Review by ChrisJWoodcock — DISMISSED** [review]

**Review by ChrisJWoodcock — APPROVED** [review]

**Pull request closed without merging**